src/overlays/actors/ovl_En_Knight/z_en_knight.h

```diff
@@ -1,29 +1,155 @@
 #ifndef Z_EN_KNIGHT_H
 #define Z_EN_KNIGHT_H
 
+#include "assert.h"
 #include "global.h"
+#include "objects/object_knight/object_knight.h"
+
+static_assert((u32)IGOS_LIMB_MAX == (u32)KNIGHT_LIMB_MAX,
+              "The skeletons for Igos du Ikana and his aides must match in structure");
 
 struct EnKnight;
 
 typedef void (*EnKnightActionFunc)(struct EnKnight*, PlayState*);
 
+typedef enum {
+    /*   0 */ EN_KNIGHT_PARAM_IGOS_MAIN = 0,
+    /*  35 */ EN_KNIGHT_PARAM_OTHERS = 35,
+    /* 100 */ EN_KNIGHT_PARAM_IGOS_HEAD = 100,
+    /* 200 */ EN_KNIGHT_PARAM_KNIGHT_BODY_AFTERIMAGE = 200,
+    /* 201 */ EN_KNIGHT_PARAM_IGOS_BODY_AFTERIMAGE,
+    /* 202 */ EN_KNIGHT_PARAM_IGOS_HEAD_AFTERIMAGE
+} EnKnightParam;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_CURTAIN_REACT_WAIT,
+    /* 1 */ KNIGHT_SUB_ACTION_CURTAIN_REACT_BEGIN,
+    /* 2 */ KNIGHT_SUB_ACTION_CURTAIN_REACT_PERFORM
+} EnKnightSubActionCurtainReact;
+
+typedef enum {
+    /* 0 */ KNIGHT_CLOSE_CURTAIN_ACTION_0,
+    /* 1 */ KNIGHT_CLOSE_CURTAIN_ACTION_1,
+    /* 2 */ KNIGHT_CLOSE_CURTAIN_ACTION_2
+} EnKnightCloseCurtainAction;
+
+typedef enum KnightBodyPart {
+    /*  0 */ KNIGHT_BODYPART_JAW,
+    /*  1 */ KNIGHT_BODYPART_TORSO,
+    /*  2 */ KNIGHT_BODYPART_PELVIS,
+    /*  3 */ KNIGHT_BODYPART_LEFT_UPPER_ARM,
+    /*  4 */ KNIGHT_BODYPART_LEFT_FOREARM,
+    /*  5 */ KNIGHT_BODYPART_SHIELD,
+    /*  6 */ KNIGHT_BODYPART_RIGHT_UPPER_ARM,
+    /*  7 */ KNIGHT_BODYPART_RIGHT_FOREARM,
+    /*  8 */ KNIGHT_BODYPART_SWORD,
+    /*  9 */ KNIGHT_BODYPART_LEFT_LEG_UPPER,
+    /* 10 */ KNIGHT_BODYPART_LEFT_LEG_LOWER,
+    /* 11 */ KNIGHT_BODYPART_LEFT_FOOT,
+    /* 12 */ KNIGHT_BODYPART_RIGHT_LEG_UPPER,
+    /* 13 */ KNIGHT_BODYPART_RIGHT_LEG_LOWER,
+    /* 14 */ KNIGHT_BODYPART_RIGHT_FOOT,
+    /* 15 */ KNIGHT_BODYPART_MAX
+} KnightBodyPart;
+
 typedef struct EnKnight {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ s16 unk_144;
-    /* 0x146 */ char unk_146[0x2];
-    /* 0x148 */ u8 unk_148;
-    /* 0x149 */ char unk_149[0x8];
-    /* 0x151 */ u8 unk_151;
-    /* 0x152 */ char unk_152[0x1];
-    /* 0x153 */ u8 unk_153;
-    /* 0x154 */ u8 unk_154;
-    /* 0x155 */ char unk_155[0x2CB];
+    /* 0x144 */ u16 curtainsLightRayAngle;
+    /* 0x146 */ s16 randTimer;
+    /* 0x148 */ u8 subAction;
+    /* 0x14A */ s16 actionTimers[3];
```

Comment:
Every other currently-documented Morita actor (Odolwa, Bio Baba, Majora, seagulls) just calls this `timers`; I think there's an implicit understanding that they are action timers from context, plus it just makes things shorter. What do you think?